### PR TITLE
GPT-Neo implementation

### PR DESCRIPTION
**Pull request by guillaume-be**
- Implementation of the GPT-Neo model
- Addition of pretrained models for GPT-Neo (125M, 1.3B and 2.7B parameters)
- Registration of GPT-Neo as a `TextGenerationPipeline` option

### CODE REVIEW

File: src/gpt_neo/gpt_neo_model.rs [hunk 1]
```diff
@@ -62,6 +72,11 @@ impl GptNeoVocabResources {
         "gpt-neo-125M/vocab",
         "https://huggingface.co/gpt2/resolve/main/vocab.json",
     );
+    /// Shared under Modified MIT license by the OpenAI team at https://github.com/openai/gpt-2/blob/master/LICENSE. Modified with conversion to C-array format.
+    pub const GPT_NEO_1_3B: (&'static str, &'static str) = (
+        "gpt-neo-1_3B/vocab",
+        "https://huggingface.co/gpt2/resolve/main/vocab.json",
```
Comment (meandmymind) [inline src/gpt_neo/gpt_neo_model.rs:88]: @guillaume-be I guess you meant gpt-neo weights here? i have a problem building this model, I guess related to wrong URLs.

Comment (guillaume-be) [reply src/gpt_neo/gpt_neo_model.rs:88]: Hi @meandmymind . This seems right - the vocabulary between the GPT-Neo models and GPT2 and identical

Comment (meandmymind) [reply src/gpt_neo/gpt_neo_model.rs:88]: @guillaume-be yes they are indeed identical. But 🤗 hosts gpt-neo vocab separately, and maybe for consistency we should use gpt-neo's url?

Comment (meandmymind) [reply src/gpt_neo/gpt_neo_model.rs:88]: Or at least add comment about using gpt2's resources as it is confusing to see gpt2's url among gpt-neo's 😀

Comment (guillaume-be) [reply src/gpt_neo/gpt_neo_model.rs:88]: Right I didn't realize that these were also uploaded to the `gpt-neo` repository.  I agree this is confusing and will push an update today. Thanks for the comments!

Comment (guillaume-be) [reply src/gpt_neo/gpt_neo_model.rs:88]: Fixed in https://github.com/guillaume-be/rust-bert/pull/146

**Pull request merged**